mdakit_membcurv/tests/test_mdakit_membcurv.py

```diff
@@ -3,10 +3,182 @@
+                                    [-0.02954713, -0.01709626, -0.01171343, -0.00766876, -0.01780511,
+                                     -0.04009694, -0.00779307, -0.04735893, 0.00799721, -0.00961057],
+                                    [-0.0033638, 0.009781, 0.06724205, 0.00795326, 0.00034145,
+                                     0.02682387, 0.10108879, -0.01423631, -0.01802192, -0.00922448],
+                                    [-0.00313899, -0.00418259, 0.03487913, -0.04456535, -0.00768992,
+                                     -0.00642677, 0.0254065, -0.01830984, -0.00904487, -0.01182518]])
+
+}
+
+
+@pytest.fixture()
+def universe():
+    u = mda.Universe(GRO_, XTC_)
+    return u
+
+
+@pytest.fixture()
+def ref_beads():
+    u = mda.Universe(GRO_PO4)
+    ref_beads = u.atoms[0:455]
+    return ref_beads
+
+
+@pytest.fixture()
+def Z_cloud(ref_beads):
+    n_cells, max_width = 10, 19
+    grid_count, grid_1, grid_2, z_ref = [np.zeros([n_cells, n_cells]) for i in range(4)]
+    factor = np.float32(n_cells / max_width)
+
+    for atom in ref_beads:
+        x, y, z = atom.position / 10
+
+        cell_l = int(abs(x) * factor)
+        cell_m = int(abs(y) * factor)
+
+        try:
+            grid_1[cell_l, cell_m] += z
+            grid_2[cell_l, cell_m] += 1
+
+        except BaseException:
+            pass
+
+    for i, j in it.product(range(n_cells), range(n_cells)):
+        if grid_2[i, j] > 0:
+            z_ref[i, j] += grid_1[i, j] / grid_2[i, j]
+            grid_count[i, j] += 1
+
+    for i, j in it.product(range(n_cells), range(n_cells)):
+        if grid_count[i, j] > 0:
+            z_ref[i, j] /= grid_count[i, j]
+
+    return z_ref
+
+
+@pytest.fixture()
+def mdtraj():
+    u = mda.Universe(GRO_, XTC_)
+    return u
+
+
+@pytest.fixture()
+def leaflets():
+    return ['lower', 'upper']
+
+
+@pytest.fixture()
+def lipid_types():
+    return ['POPC', 'POPE']
+
 
 def test_mdakit_membcurv_imported():
     """Sample test, will always pass so long as import statement worked"""
     assert "mdakit_membcurv" in sys.modules
+
+
+def test_output_folder():  # line 76, gone after refactoring
+    tmp_dir = 'tmp_output/'
+    os.makedirs(os.path.dirname(tmp_dir), exist_ok=True)
+    assert os.path.exists(tmp_dir)
+    os.rmdir(tmp_dir)
+
+
+def test_md_topology():  # line 92, gone after refactoring
+    top = md.load(GRO_).topology
+    assert top.n_atoms == 18891
+
+
+def test_box_size(universe):  # line 99
+    box_size = universe.dimensions[0]
+    assert_almost_equal(box_size, 184.31013, decimal=5)
+
+
+def test_max_width(universe):  # line 100
+    max_width = universe.dimensions[0] * 0.1
+    assert max_width == 18.431013488769533
+
+
+def test_n_cells(universe):  # line 101
+    unit_width = 20
+    max_width = universe.dimensions[0] * 0.1
+    n_cells = math.ceil(max_width / unit_width * 10)
+    print(n_cells)
+    assert n_cells == 10.
```

Comment:
These tests are very fine-grained. `test_box_size` is mostly checking your assumptions of how MDAnalysis reads dimensions, which is fine, but `test_max_width` seems to be testing that division by 10 works as expected -- I think that is probably unnecessary.

In general the tests in this repository should be for *your* code. A good rule of thumb is that each test should test the output of a function that you wrote. So if you had a function called `get_max_width`, you could test that, but IMO it's out of scope to test `MDAnalysis` and `math` functionality.

Reply:
Thanks. 
Maybe I need more directions at this point. I remembered I asked @orbeckst about what part of the code should I test to start, considering I have a `main.py` and `mods.py`. I tried to test main, with the exception of the `parse_arg` part and some other redundant functions like `curvature` that I wrote just to made my life easier when I was using it for me. 
I thought what I had in this version of the tests was checking some lines in `main.py` and some other is `mods.py`. Not all of them, because there are parts of the code that will be gone in the refactoring. 
Maybe mentors can help giving me some lines to test? ☺️

Reply:
Maybe add at least one test for every function that you think you will need in some form in the MDAnalysis refactor? Even if the form of the function changes when refactoring, we would want to check that the returned output for the given input does not.